src/lib/y2network/interface.rb

```diff
@@ -34,6 +34,7 @@ def initialize(name)
     # @param other [Interface] Interface to compare with
     # @return [Boolean]
     def ==(other)
+      return false if self.class != other.class
```

Comment:
well, I still like more my code as this one basically means that you kill inheritance. using `is_a?` allows to have child of interface that can be also compared.

Reply:
Sorry, somehow I overlooked the suggestion. Applied.